# Initiation Python

Ce premier TP a pour but de vous aider à la prise en main d'une part, de Jupyter Notebook, mais également de la logique de programmation en Python.  

Pourquoi Python ?
 - langage de haut niveau, concis et très général
 - nombreuses librairies, notamment pour l'Intelligence Artificielle ou le Deep Learning
 - populaire 
 - libre et gratuit, avec une bonne communauté

## Langage interprété

Python est un langage interprété, ce qui signifie qu'on peut évaluer des expressions seules en exécutant (avec le bouton "Executer" ou CTRL + ENTER) la cellule concernée

Exécutez les cellules suivantes et observez les sorties :

In [ ]:
10*6+18

In [ ]:
a = "je pense donc je suis"

In [ ]:
a

In [ ]:
print(a)

In [ ]:
b = 40

In [ ]:
b * 10 - 25

In [9]:
liste=list(range(10))

In [ ]:
liste[2:4]

In [ ]:
liste[2:6:2]

In [ ]:
liste[:4]

In [ ]:
liste[::2]

In [12]:
liste.append(2)

In [ ]:
liste.pop(3)

In [14]:
del(liste[2])

In [16]:
liste2=[x**2 for x in liste]

In [17]:
liste3=[x**2 for x in liste if x%2==0]

In [ ]:
reponse=input("Gauche ou Droite ? ")

In [ ]:
reponse

## Explorer une base de données

Nous allons étudier une base de données très simple et très connue : IRIS. Cette BDD contient les informations de 3 variétés de fleurs sur leurs longueurs et largeurs, de sépal et de pétales.  
Ce premier exercice vous permettra de manipuler les données et de bien appréhender le langage Python.  
Comme pour tous les langages, vous trouverez énormément de documentations et d'aide en ligne, n'hésitez pas à chercher et tester des choses.

1. Import des données

In [14]:
!pip install pydataset #à faire uniquement la première fois, pour télécharger le package sur sa machine

  Created wheel for pydataset: filename=pydataset-0.2.0-py3-none-any.whl size=15939434 sha256=bbba89daabfd4b3976432edfbe67a23a0f496107862e4b01cf4c4c858dceb83d
  Stored in directory: c:\users\aliso\appdata\local\pip\cache\wheels\d7\e5\36\85d319586b4a405d001029d489102f526ce5546248c295932a
Successfully built pydataset


In [15]:
from pydataset import data
iris = data("iris")

initiated datasets repo at: C:\Users\aliso\.pydataset/


In [16]:
iris

Sepal.Length  Sepal.Width  Petal.Length  Petal.Width     Species
1             5.1          3.5           1.4          0.2      setosa
2             4.9          3.0           1.4          0.2      setosa
3             4.7          3.2           1.3          0.2      setosa
4             4.6          3.1           1.5          0.2      setosa
5             5.0          3.6           1.4          0.2      setosa
6             5.4          3.9           1.7          0.4      setosa
7             4.6          3.4           1.4          0.3      setosa
8             5.0          3.4           1.5          0.2      setosa
9             4.4          2.9           1.4          0.2      setosa
10            4.9          3.1           1.5          0.1      setosa
11            5.4          3.7           1.5          0.2      setosa
12            4.8          3.4           1.6          0.2      setosa
13            4.8          3.0           1.4          0.1      setosa
14            4.3          3.0           1.1          0.1      setosa
15            5.8          4.0           1.2          0.2      setosa
16            5.7          4.4           1.5          0.4      setosa
17            5.4          3.9           1.3          0.4      setosa
18            5.1          3.5           1.4          0.3      setosa
19            5.7          3.8           1.7          0.3      setosa
20            5.1          3.8           1.5          0.3      setosa
21            5.4          3.4           1.7          0.2      setosa
22            5.1          3.7           1.5          0.4      setosa
23            4.6          3.6           1.0          0.2      setosa
24            5.1          3.3           1.7          0.5      setosa
25            4.8          3.4           1.9          0.2      setosa
26            5.0          3.0           1.6          0.2      setosa
27            5.0          3.4           1.6          0.4      setosa
28            5.2          3.5           1.5          0.2      setosa
29            5.2          3.4           1.4          0.2      setosa
30            4.7          3.2           1.6          0.2      setosa
31            4.8          3.1           1.6          0.2      setosa
32            5.4          3.4           1.5          0.4      setosa
33            5.2          4.1           1.5          0.1      setosa
34            5.5          4.2           1.4          0.2      setosa
35            4.9          3.1           1.5          0.2      setosa
36            5.0          3.2           1.2          0.2      setosa
37            5.5          3.5           1.3          0.2      setosa
38            4.9          3.6           1.4          0.1      setosa
39            4.4          3.0           1.3          0.2      setosa
40            5.1          3.4           1.5          0.2      setosa
41            5.0          3.5           1.3          0.3      setosa
42            4.5          2.3           1.3          0.3      setosa
43            4.4          3.2           1.3          0.2      setosa
44            5.0          3.5           1.6          0.6      setosa
45            5.1          3.8           1.9          0.4      setosa
46            4.8          3.0           1.4          0.3      setosa
47            5.1          3.8           1.6          0.2      setosa
48            4.6          3.2           1.4          0.2      setosa
49            5.3          3.7           1.5          0.2      setosa
50            5.0          3.3           1.4          0.2      setosa
51            7.0          3.2           4.7          1.4  versicolor
52            6.4          3.2           4.5          1.5  versicolor
53            6.9          3.1           4.9          1.5  versicolor
54            5.5          2.3           4.0          1.3  versicolor
55            6.5          2.8           4.6          1.5  versicolor
56            5.7          2.8           4.5          1.3  versicolor
57            6

2. Afficher les données

Utiliser les fonctions head() et tail() pour afficher les 5 premières et 5 dernières lignes du dataset

Sepal.Length  Sepal.Width  Petal.Length  Petal.Width Species
1           5.1          3.5           1.4          0.2  setosa
2           4.9          3.0           1.4          0.2  setosa
3           4.7          3.2           1.3          0.2  setosa
4           4.6          3.1           1.5          0.2  setosa
5           5.0          3.6           1.4          0.2  setosa

Sepal.Length  Sepal.Width  Petal.Length  Petal.Width    Species
146           6.7          3.0           5.2          2.3  virginica
147           6.3          2.5           5.0          1.9  virginica
148           6.5          3.0           5.2          2.0  virginica
149           6.2          3.4           5.4          2.3  virginica
150           5.9          3.0           5.1          1.8  virginica

3. Quel est la taille du dataset ? Utiliser la fonction shape et interpréter le résultat

(150, 5)

4. Afficher toutes les colonnes du dataset

Index(['Sepal.Length', 'Sepal.Width', 'Petal.Length', 'Petal.Width',
       'Species'],
      dtype='object')

5. Sélectionner uniquement la colonnes Sepal.Length 

1      5.1
2      4.9
3      4.7
4      4.6
5      5.0
6      5.4
7      4.6
8      5.0
9      4.4
10     4.9
11     5.4
12     4.8
13     4.8
14     4.3
15     5.8
16     5.7
17     5.4
18     5.1
19     5.7
20     5.1
21     5.4
22     5.1
23     4.6
24     5.1
25     4.8
26     5.0
27     5.0
28     5.2
29     5.2
30     4.7
31     4.8
32     5.4
33     5.2
34     5.5
35     4.9
36     5.0
37     5.5
38     4.9
39     4.4
40     5.1
41     5.0
42     4.5
43     4.4
44     5.0
45     5.1
46     4.8
47     5.1
48     4.6
49     5.3
50     5.0
51     7.0
52     6.4
53     6.9
54     5.5
55     6.5
56     5.7
57     6.3
58     4.9
59     6.6
60     5.2
61     5.0
62     5.9
63     6.0
64     6.1
65     5.6
66     6.7
67     5.6
68     5.8
69     6.2
70     5.6
71     5.9
72     6.1
73     6.3
74     6.1
75     6.4
76     6.6
77     6.8
78     6.7
79     6.0
80     5.7
81     5.5
82     5.5
83     5.8
84     6.0
85     5.4
86     6.0
87     6.7
88     6.3
89     5.6
90     5.5
91     5.5

6. Regarder le nombre de fleurs par variétés que contient le dataset (utiliser la fonction value_counts() )

setosa        50
virginica     50
versicolor    50
Name: Species, dtype: int64

7. Quelle est la largeur moyenne du sépal ?

8. Quel est la valeur maximale de longueur de pétale ?

9. Sélectionner uniquement les données pour lesquelles la largeur du pétal est >= 5.1 et stocker le résultat dans une nouvelle variable